plugins/catalog-backend/src/ingestion/processors/OwnerRelationProcessor.ts

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Entity,
+  ENTITY_DEFAULT_NAMESPACE,
+  LocationSpec,
+  parseEntityRef,
+  ApiEntityV1alpha1,
+  ComponentEntityV1alpha1,
+  RELATION_OWNED_BY,
+  RELATION_OWNER_OF,
+} from '@backstage/catalog-model';
+import { CatalogProcessor, CatalogProcessorEmit } from './types';
+import * as result from './results';
+
+const includedKinds = new Set(['api', 'component']);
+
+export class OwnerRelationProcessor implements CatalogProcessor {
```

Comment:
yeah as i mentioned elsewhere, i wonder if it would be interesting to pursue the ability to make

```ts
class ComponentEntityProcessor implements CatalogProcessor {
  async validateKind(entity: Entity): boolean {
    // return false if not a known kind/apiVersion, otherwise validate and either throw or return true
    // these are called in a oneOf fashion by the LocationReader
  }

  // ... other concerns around that same kind, such as relation extraction
}
```

... to keep it all in one easy-to-find place. And you would no longer specify any kind policies to the CatalogBuilder, but instead entire processors that keep all concerns together

Reply:
Right, although this isn't really a good example of that right? as we're processing both API and Component entities

Reply:
Maybe we shouldn't :)

Or alternatively, if we invert control and make a plugin system for the backend, we'd have a single `DefaultEntityKindsPlugin` that adds several things to the processing slots as it sees fit (as one "hook" or several, depending on how we prefer to do it)